pkg/agent/nplagent/k8s/annotations.go

```diff
@@ -0,0 +1,164 @@
+	var err error
+	current := make(map[string]string)
+	if pod.Annotations != nil {
+		current = pod.Annotations
+	}
+
+	klog.Infof("Building annotation for pod: %s\tport: %s --> %s:%s", pod.Name, containerPort, nodeIP, nodePort)
+
+	var annotations []NPLEPAnnotation
+	// nplEP annotation exists
+	if current[NPLAnnotationStr] != "" {
+		if err = json.Unmarshal([]byte(current[NPLAnnotationStr]), &annotations); err != nil {
+			klog.Warningf("Unable to unmarshal NPLEP annotation")
+		}
+
+		if !IsNodePortInAnnotation(annotations, nodePort) {
+			annotations = append(annotations, NPLEPAnnotation{
+				PodPort:  containerPort,
+				NodeIP:   nodeIP,
+				NodePort: nodePort,
+			})
+		} else {
+			// mapping for the containerPort already exists
+			// TODO
+		}
+	} else {
+		annotations = []NPLEPAnnotation{NPLEPAnnotation{
+			PodPort:  containerPort,
+			NodeIP:   nodeIP,
+			NodePort: nodePort,
+		}}
+	}
+
+	current[NPLAnnotationStr] = Stringify(annotations)
+	pod.Annotations = current
+}
+
+func removeFromPodAnnotation(pod *corev1.Pod, containerPort string) {
+	var err error
+	current := pod.Annotations
+
+	klog.Infof("Removing annotation from pod: %s\tport: %s", pod.Name, containerPort)
+	var annotations []NPLEPAnnotation
+	if err = json.Unmarshal([]byte(current[NPLAnnotationStr]), &annotations); err != nil {
+		klog.Warningf("Unable to unmarshal NPLEP annotation")
+		return
+	}
+
+	for i, ann := range annotations {
+		if ann.PodPort == containerPort {
+			annotations = append(annotations[:i], annotations[i+1:]...)
+			break
+		}
+	}
+
+	current[NPLAnnotationStr] = Stringify(annotations)
+	pod.Annotations = current
+}
+
+// RemoveNPLAnnotationFromPods : Removes npl annotations from all pods
+func (c *Controller) RemoveNPLAnnotationFromPods() {
+	podList, err := c.KubeClient.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Warningf("Unable to list Pods")
+		return
+	}
+	nodeName, err := env.GetNodeName()
+	if err != nil {
+		klog.Warningf("Failed to get nodename, NPL annotation can not be removed for pods")
+		return
+	}
+	for _, pod := range podList.Items {
+		if nodeName != pod.Spec.NodeName {
+			continue
+		}
+		podAnnotation := pod.GetAnnotations()
+		if podAnnotation == nil {
+			continue
+		}
+		klog.Infof("Removing all NPL annotation from pod: %s, ns: %s", pod.Name, pod.Namespace)
+		delete(podAnnotation, NPLAnnotationStr)
+		pod.Annotations = podAnnotation
+		c.KubeClient.CoreV1().Pods(pod.Namespace).Update(context.TODO(), &pod, metav1.UpdateOptions{})
+	}
+}
+
+func (c *Controller) updatePodAnnotation(pod *corev1.Pod) error {
+	if _, err := c.KubeClient.CoreV1().Pods(pod.Namespace).Update(context.TODO(), pod, metav1.UpdateOptions{}); err != nil {
+		klog.Warningf("Unable to update pod %s with annotation: %+v", pod.Name, err)
+		return err
+	}
+	klog.Infof("Successfully updated pod %s %s annotation", pod.Name, pod.Namespace)
```

Comment:
```suggestion
	klog.Infof("Successfully updated pod %s/%s annotation", pod.Namespace, pod.Name)
```
to be more readable and natural?

Reply:
@tnqn I have updated incorporating the changes you suggested. Please let me know further comments for the other files.